src/software/ai/navigator/path_planner/enlsvg_path_planner.cpp

```diff
@@ -0,0 +1,199 @@
+    : resolution(resolution),
+      num_grid_rows(static_cast<int>(round(navigable_area.xLength() / resolution))),
+      num_grid_cols(static_cast<int>(round(navigable_area.yLength() / resolution))),
+      origin(navigable_area.negXNegYCorner()),
+      max_navigable_y_enlsvg_point(
+          convertPointToEnlsvgPoint(navigable_area.posXPosYCorner()).y),
+      max_navigable_x_enlsvg_point(
+          convertPointToEnlsvgPoint(navigable_area.posXPosYCorner()).x),
+      grid(std::make_unique<EnlsvgGrid>(num_grid_rows, num_grid_cols))
+{
+    createObstaclesInGrid(obstacles, grid_boundary_offset);
+    algo = std::make_unique<const EnlsvgAlgorithm>(*grid);
+    mem  = std::make_unique<EnlsvgMemory>(*algo);
+}
+
+void EnlsvgPathPlanner::createObstaclesInGrid(const std::vector<ObstaclePtr> &obstacles,
+                                              double boundary_margin) const
+{
+    double offset_in_enlsvg = boundary_margin / resolution;
+    for (int x = 0; x < offset_in_enlsvg; ++x)
+    {
+        for (int y = 0; y < num_grid_cols; ++y)
+        {
+            grid->setBlocked(x, y, true);
+            grid->setBlocked(num_grid_rows - 1 - x, y, true);
+        }
+    }
+    for (int y = 0; y < offset_in_enlsvg; ++y)
+    {
+        for (int x = 0; x < num_grid_rows; ++x)
+        {
+            grid->setBlocked(x, y, true);
+            grid->setBlocked(x, num_grid_cols - 1 - y, true);
+        }
+    }
+
+    for (auto &obstacle : obstacles)
+    {
+        auto blocked_points = obstacle->rasterize(resolution);
+
+        for (Point &blocked_point : blocked_points)
+        {
+            EnlsvgPoint blocked_coord = convertPointToEnlsvgPoint(blocked_point);
+            if (isCoordNavigable(blocked_coord))
+            {
+                grid->setBlocked(blocked_coord.x, blocked_coord.y, true);
+            }
+        }
+    }
+}
+
+bool EnlsvgPathPlanner::isCoordNavigable(const EnlsvgPoint &ep) const
+{
+    return (ep.x >= 0 && ep.x < max_navigable_x_enlsvg_point) &&
+           (ep.y >= 0 && ep.y < max_navigable_y_enlsvg_point);
+}
+
+std::optional<Path> EnlsvgPathPlanner::findPath(const Point &start, const Point &end,
+                                                const Rectangle &navigable_area,
+                                                const std::vector<ObstaclePtr> &ignored)
+{
+    // Check if start and end coordinates are in navigable area and return null if it
+    // isn't
+    bool navigable_area_contains_start =
+        (start.x() >= navigable_area.xMin()) && (start.x() <= navigable_area.xMax()) &&
+        (start.y() >= navigable_area.yMin()) && (start.y() <= navigable_area.yMax());
+    bool navigable_area_contains_end =
+        (end.x() >= navigable_area.xMin()) && (end.x() <= navigable_area.xMax()) &&
+        (end.y() >= navigable_area.yMin()) && (end.y() <= navigable_area.yMax());
+    if (!navigable_area_contains_start || !navigable_area_contains_end)
+    {
+        return std::nullopt;
+    }
```

Comment:
can you use `!navigable_area.contains(start) || !navigable_area.contains(end)` here?

Reply:
did we want to avoid using `contains` in the path planner as a performance optimization @nimazareian ? I remember this was part of the path planner performance improvements for theta star earlier in the year

Reply:
I don't think this would give any significant performance boost. The implementation is pretty much the same as `contains` anyways. This change was also made before the performance improvements PR, [here](https://github.com/UBC-Thunderbots/Software/commit/894e7f0397cdaf5d0fc5d85cf888556ff053ebd2).
I personally don't see a reason against using `contains`.

Reply:
fixed